packages/lib/passport-interface/src/SubscriptionManager.ts

```diff
@@ -278,34 +293,66 @@ export class FeedSubscriptionManager {
     const responses: SubscriptionActions[] = [];
     this.resetError(subscription.id);
     try {
-      const authKey = await this.getAuthKeyForFeed(subscription);
+      const authKey = undefined; //await this.getAuthKeyForFeed(subscription);
 
-      const pcdCredential: SerializedPCD | undefined = authKey
+      const pcdCredentials: SerializedPCD[] | undefined = authKey
         ? await this.makeAlternateCredentialPCD(authKey)
-        : await credentialManager.requestCredential({
+        : await credentialManager.requestCredentials({
             signatureType: "sempahore-signature-pcd",
             pcdType: subscription.feed.credentialRequest.pcdType
           });
 
-      const result = await this.api.pollFeed(subscription.providerUrl, {
-        feedId: subscription.feed.id,
-        pcd: pcdCredential
-      });
+      const results = await Promise.all(
+        pcdCredentials.map((credential) => {
+          return this.api.pollFeed(subscription.providerUrl, {
+            feedId: subscription.feed.id,
+            pcd: credential
+          });
+        })
+      );
 
-      if (!result.success) {
-        if (result.code === 410) {
-          this.flagSubscriptionAsEnded(subscription.id, result.error);
+      const failure = results.find((r) => !r.success);
+      if (failure?.success === false) {
+        if (failure.code === 410) {
+          this.flagSubscriptionAsEnded(subscription.id, failure.error);
           return responses;
         }
 
-        throw new Error(result.error);
+        throw new Error(failure.error);
       }
 
-      const { actions } = result.value;
+      const actions = results.flatMap((r) => {
+        if (r.success) {
+          return r.value.actions;
+        }
+        return [];
+      });
+
+      // in the case that we're requesting from the same feed
+      // multiple times using two different email credentials,
+      // the result of one of the requests would overwrite the
+      // result of the other request, given both have a 'delete'
+      // action. thus, we coalesce equivalent delete folder actions
+      // into the the first occurrence of a deletion for each folder.
+      const filteredActions = [];
```

Comment:
This feels fragile.  There's also a cost to the extra round-trips for each email.
Have you considered the alternative of sending all emails to the server in a single request, so the server can merge together the list of PCDs as it wishes, and send them all along with a single delete action.

Reply:
> Have you considered the alternative of sending all emails to the server in a single request

I've done this instead, and removed this change.